lib/AST/DeclContext.cpp

```diff
@@ -475,6 +475,25 @@ unsigned DeclContext::getSemanticDepth() const {
   return 1 + getParent()->getSemanticDepth();
 }
 
+bool DeclContext::mayContainMembersAccessedByDynamicLookup() const {
+  // Dynamic lookup can only find class and protocol members, or extensions of
+  // classes.
+  if (auto *NTD = dyn_cast<NominalTypeDecl>(this)) {
```

Comment:
In reviewing this code, I wanted to read this function and ensure it was doing what the comment implied. But it defeats me, in my code-reviewing mode. Could you rewrite or refactor it to make it clearer? For instance, the flow can reach 485 if it's a ClassDecl, so why check for ProtocolDecl?

Would it work to just start with a check `isGenericContext` and return false if it is generic?

Reply:
Which part is unclear? Only one of line 482 or line 485 will test true. We're checking for class members, protocol members, and finally class extension members.

Reply:
We discussed this just now offline. Your revised version (future PR?) is great